R/plot.R

```diff
@@ -68,57 +68,196 @@ fldts2df <- function(fldts, griddata)
+                               axis.text=element_blank(),
+                               axis.ticks=element_blank(),
+                               legend.position = 'bottom') +
+                # black outlines of countries
+                ggplot2::geom_path(data = countries,
+                                   aes(x = long, y = lat, group = group),
+                                   size = 0.25, color = 'black')
+        }
+
+}
+
+
+#' Plot a single year's temperature for every generated gridded time series
+#' in a list.
+#'
+#' Takes in a list of temperature fields (residual or full data), a year
+#' index, and the underlying temperature gridded data from a trained
+#' emulator, as well as a few arguments relevant to plotting
+#'
+#' @param fieldlist List of residual
+#' @param yearindex Index of the year to be plotted. For example, in data
+#' ranging from 2006 to 2100, the yearindex for 2006 is 1.
+#' @param emulatorgriddata The griddataT structure returned in a trained
+#' emulator.
+#' @param minval Lower limit of the color scale.  The default value was chosen
+#' to work well for fields of residuals from the mean temperature response.
+#' @param maxval Upper limit of the color scale.  The default value was chosen
+#' to work well for fields of residuals from the mean temperature response.
+#' @param legendstr String to use for the title of the graph legend
+#' @param palettestr String with the name of the ggplot2 color palette to use.
+#' Defaults to 'RdYlBu'. Options are Diverging: 'BrBG', 'PiYG', 'PRGn', 'PuOr',
+#' 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn', 'Spectral'; Qualitative: 'Accent',
+#' 'Dark2', 'Paired', 'Pastel1', 'Pastel2', 'Set1', 'Set2', 'Set3';
+#' Sequential: 'Blues', 'BuGn', 'BuPu', 'GnBu', 'Greens', 'Greys', 'Oranges',
+#' 'OrRd', 'PuBu', 'PuBuGn', 'PuRd', 'Purples', 'RdPu', 'Reds', 'YlGn',
+#' 'YlGnBu', 'YlOrBr', 'YlOrRd'.
+#' @param palettedir Numerical value, only accepts 1 or -1, determining the
+#' order of the color palette used. Defaults to -1.
+#'
+#' @author ACS May 2020
+#' @export
+plotglobalfieldsT <- function(fieldlist, yearindex, emulatorgriddata,
+                              minval=-3.5, maxval=3.5, legendstr,
+                              palettestr = 'RdYlBu', palettedir = -1) {
+
+
+    Ngrid <- ncol(emulatorgriddata[[1]])
+
+    ## Extract a single example field from each series (yearindex determines which one) and create a plot
+    lapply(fieldlist, function(g) {
+        suppressWarnings(
+            plot_field(g[yearindex, 1:Ngrid], emulatorgriddata, 14,  # 14 is the number of color levels in the plot
+                       minval, maxval,
+                       legendstr = legendstr, palettestr = palettestr, palettedir = palettedir) +
+                guides(title=legendstr, title.position='top', title.hjust=0.5)
+        )
+    })
+}
+
+
+
+#' Plot a single year's precipitation for every generated gridded time series
+#' in a list.
+#'
```

Comment:
Why are the precip and temp functions different from one another? Could they be a single function? Could you add documentation to explain why they are not interchangeable?